### PR TITLE
[65449] Adding body to Sign up Service agreements request

### PR DESCRIPTION
**Pull request by bosawt**
## Summary

- This PR adds the body data to the Sign Up Service `POST agreements` request. This is done to align the request itself with what Sign Up Service is expecting for a terms of use agreements update

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#65449

## Testing done

- Made a call to `MobileApplicationPlatform::SignUp::Service.new.agreements_accept(icn:, signature_name:, version:)` and took the resulting cURL, then used that cURL in staging with a relevant access token to prove it was a proper agreements_accept call

## What areas of the site does it impact?
Terms of Use

## Acceptance criteria

- [ ] You'll have to extract a curl from the Sign Up Service, so add `conn.request(:curl, ::Logger.new(STDOUT), :warn)` to the faraday connection in `sign_up/configuration.rb`
- [ ] In `sign_up/service.rb`, hack `authenticated_header`, so that the `access_token` variable is `{ access_token: 'some-access-token }`
- [ ] Make a proper call to the agreements accept endpoint: `MobileApplicationPlatform::SignUp::Service.new.agreements_accept(icn:, signature_name:, version:)`
- [ ] In your log output, you should see a cURL, take that cURL, and replace the access token with a access token call on a staging instance: `MobileApplicationPlatform::SecurityToken::Service.new.token(application: :sign_up_service, icn:)`
- [ ] After submitting the cURL, you should see either a good response, or something like this: `{"id":"3e8d6c72-f471-432b-bc09-6e768d91520a","code":400,"message":"Missing ICN Identifier","errorCode":10,"detail":"JWT must include an ICN”}` (this is an error on the Sign Up Service side that they are currently resolving)

### CODE REVIEW

File: app/services/terms_of_use/acceptor.rb [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 module TermsOfUse
   class Acceptor
-    def initialize(user_account:, version:)
+    def initialize(user_account:, common_name:, version:)
```
Comment (LindseySaari) [inline app/services/terms_of_use/acceptor.rb:5]: It looks like this file doesn't have a CODEOWNERS entry

Comment (LindseySaari) [reply app/services/terms_of_use/acceptor.rb:5]: Would you mind rebasing from master. I made a change to [cover both cases for CODEOWNERS directory definitions here](https://github.com/department-of-veterans-affairs/vets-api/pull/13803)

**Comment by LindseySaari** [comment]
@bosawt I made a few changes to my grep logic. I merged in master to this PR

**Comment by LindseySaari** [comment]
@bosawt It looks like app/workers/terms_of_use/sign_up_service_updater_job.rb is missing. I see you have app/jobs/terms_of_use/ but notice the file is app/worker

**Comment by bosawt** [comment]
@LindseySaari Looks like with your changes, and the fix that you identifier with my previous CODEOWNERS definitions, this should be good to go now

**Review by joeniquette — APPROVED** [review]

**Review by joeniquette — DISMISSED** [review]

**Comment by bosawt** [comment]
@LindseySaari When you get a chance, can you re-review? I fixed the merge conflicts with CODEOWNERS

File: lib/mobile_application_platform/sign_up/configuration.rb [hunk 2]
```diff
@@ -42,6 +42,13 @@ def provisioning_acceptable_status
         [200, 406, 412]
       end
 
+      def agreements_version_mapping
+        {
+          'v1' => 2,
+          'v2' => 3
+        }
+      end
```
Comment (dickdavis) [inline lib/mobile_application_platform/sign_up/configuration.rb:50]: Would this agreement mapping make more sense to be included on the `TermsOfUseAgreement` model? I can see us needing this mapping in other process flows

File: lib/mobile_application_platform/sign_up/service.rb [hunk 3]
```diff
@@ -61,6 +64,15 @@ def authenticated_header(icn)
         config.authenticated_header(access_token[:access_token])
       end
 
+      def agreements_body(icn, signature_name, version)
+        {
+          responseDate: Time.zone.now,
+          icn:,
+          signatureName: signature_name,
+          version: config.agreements_version_mapping[version]
```
Comment (dickdavis) [inline lib/mobile_application_platform/sign_up/service.rb:72]: If you moved the mapping to the model, you could do something like

```ruby
version: TermsOfUseAgreement.version_map[version]
```

**Review by dickdavis — DISMISSED** [review]
LGTM, one question, but no blockers. Can't test, so consider my approval supplemental to Joe's.

**Review by LindseySaari — APPROVED** [review]

**Review by LindseySaari — APPROVED** [review]

File: lib/mobile_application_platform/sign_up/configuration.rb (hunk 2 above)
Comment (bosawt) [reply lib/mobile_application_platform/sign_up/configuration.rb:50]: This one's a good point, I'm still a little on the fence with there to put it though. Basically we have our own internal versions (`v1`, `v2`), and the SignUpService has mappings that we only care about with regards to our interface with SignUpService. So I'm not actually sure if we will ever need this mapping outside of the `SignUp::Service` class... 

But on the other hand it's kinda weird to put information in the service config that's not relevant directly to the service (the `v1`, `v2` part). I suppose we've performed this straddling of concerns in the past by just making a `Constants` class, so maybe that's the right answer here... I'll let us refactor after this PR if that seems like a good solution

**Pull request merged**